engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/KafkaSingleScenarioTaskRun.scala

```diff
@@ -125,9 +125,21 @@ class KafkaSingleScenarioTaskRun(taskId: String,
   }
 
   private def sendOutputToKafka(output: ResultType[interpreterTypes.EndResult[ProducerRecord[Array[Byte], Array[Byte]]]]): Future[_] = {
-    val results = output.value.map(_.result)
+
+    val resultsWithEventTimestamp = output.value.map(endResult => {
+      val contextEventTimestamp = endResult.context.variables(VariableConstants.EventTimestampVariableName).asInstanceOf[Long]
```

Comment:
`endResult.context.get[Long]` - I think we should be prepared for the case where the variable is missing and just leave null (`orElse(contextEventTimestamp).orNull`) in line 135

Reply:
fixed it, thx